pyodk/endpoints/comments.py

```diff
@@ -0,0 +1,113 @@
+import logging
+from datetime import datetime
+from typing import List, Optional
+
+from pyodk import validators as pv
+from pyodk.endpoints import bases
+from pyodk.errors import PyODKError
+from pyodk.session import Session
+
+log = logging.getLogger(__name__)
+
+
+class Comment(bases.Model):
+    body: str
+    actorId: int
+    createdAt: datetime
+
+
+class URLs(bases.Model):
+    class Config:
+        frozen = True
+
+    list: str = "projects/{project_id}/forms/{form_id}/submissions/{instance_id}/comments"
+    post: str = "projects/{project_id}/forms/{form_id}/submissions/{instance_id}/comments"
+
+
+class CommentService(bases.Service):
```

Comment:
There's a broader question around how to structure access to nested resources or resources that are not as central to Central as `projects`, `forms`, `submission`. This intersects a bit with the questions I raised around users at https://github.com/getodk/pyodk/issues/7 

In my mind a goal is to match user expectations and common usage over strictly mapping to the Central API. I think where we do match Central we want to match the frontend user experience more than the backend API.

Comments feel very much connected to submissions. What do you think of `client.submission.list_comments`, `client.submission.add_comment`? I realize this means a looser internal structure and maybe moving away from the `endpoints` package name.

Reply:
I'm OK with the name/structure as-is but please let me know if you'd like to change it. There could be a higher level module/package that holds all these convenience methods, rather than mixing them into the endpoint mappings.

Reply:
Marked most of pyodk as private (package name prefixed with underscore, and in top-level `__init__.py` specified public elements in `__all__`). Exceptions are `client.Client` and `errors` module. On `client.Client`, the `_comments` property is marked private. All the docs/examples only deal with the `Client`.

Added `list_comments` and `add_comment` methods to `Submission`. These just delegate to the existing `Comments` module. So given the above, the internal structure can be clean but what is exposed is as described above.